R/zchunk_LB123.Electricity_china.R

```diff
@@ -0,0 +1,262 @@
+      mutate(fuel = replace(fuel, fuel == "oil" & province == "XZ", "refined liquids"),
+             fuel = replace(fuel, fuel == "natural.gas" & province == "XZ", "gas"),
+             # In Tibet our oil and gas numbers are in electricity out and they need to be energy in in Mtce. This is a total hack but
+             # just divide by the efficiency. Gas is 0.55 and oil is about 0.35
+             value =  replace(value, province == "XZ" & fuel %in% c("refined liquids", "gas"), 0),
+             copy_rl = value * 100 * CONV_GWH_EJ * CONV_EJ_MTCE / -0.35,
+             copy_g = value * 100 * CONV_GWH_EJ * CONV_EJ_MTCE / -0.55,
+             copy_rl = replace(value, province == "XZ" & fuel == "refined liquids", 0),
+             copy_g = replace(value, province == "XZ" & fuel == "gas", 0),
+             value = value + copy_rl + copy_g) %>%
+      select(-copy_g, -copy_rl) %>%
+      # Aggregate by fuel compute each province's percentage, by fuel
+      group_by(fuel, year) %>%
+      mutate(pct = value / sum(value)) %>%
+      ungroup %>%
+      mutate(fuel = replace(fuel, fuel == 'solar', 'solar PV')) %>%
+      select(-value) %>%
+      replace_na(list(pct = 0)) ->
+      L123.pct_province_elec_F
+
+    # Electricity generation inputs by fuel and province
+    # Note that fuel inputs are only available for selected fuels; province_share_data uses this relevant subset
+    L123.in_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID, fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.in_EJ_province_elec_F
+    Fuel_in_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
+    L123.pct_province_elec_F %>%
+      filter(fuel %in% Fuel_in_Yh & year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L123.in_EJ_province_elec_F, by = c('year', 'fuel')) %>%
+      mutate(elec_F = value * pct) %>%
+      select(-value, -pct, -GCAM_region_ID) ->
+      L123.in_EJ_province_elec_F
+
+    # Electricity generation outputs by fuel and province
+    L123.out_EJ_R_elec_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID, fuel %in% L123.pct_province_elec_F$fuel) ->
+      L123.out_EJ_province_elec_F
+    Fuel_out_Yh = unique(L123.in_EJ_R_elec_F_Yh$fuel)
+    L123.pct_province_elec_F %>%
+      filter(fuel %in% Fuel_out_Yh, year %in% HISTORICAL_YEARS) %>%
+      left_join_error_no_match(L123.out_EJ_province_elec_F, by = c('year', 'fuel')) %>%
+      mutate(elec_F = value * pct) %>%
+      select(-value, -pct, -GCAM_region_ID) ->
+      L123.out_EJ_province_elec_F
+
+    # ELECTRICITY - OWNUSE
+    # Electricity own use by province
+    L126.in_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.in_EJ_CHINA_ownuse
+
+    L126.out_EJ_R_elecownuse_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) ->
+      L123.out_EJ_CHINA_ownuse
+
+    L123.in_EJ_CHINA_ownuse %>%
+      left_join_error_no_match(L123.out_EJ_CHINA_ownuse, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%
```

Comment:
Remove line 135-141, and change line 142 as:
`left_join_error_no_match(L126.out_EJ_R_elecownuse_F_Yh, by = c("GCAM_region_ID", "sector", "fuel", "year")) %>%`